### PR TITLE
Add link to QUIC prerequisities and supported operating systems

### PR DESCRIPTION
**Pull request by jkotas**
Fixes https://github.com/dotnet/runtime/issues/80001

### CODE REVIEW

File: xml/System.Net.Quic/QuicConnection.xml [hunk 1]
```diff
@@ -28,6 +28,8 @@ A <xref:System.Net.Quic.QuicConnection> can either be accepted from <xref:System
 
 Each connection can then open an outbound stream using <xref:System.Net.Quic.QuicConnection.OpenOutboundStreamAsync(System.Net.Quic.QuicStreamType,System.Threading.CancellationToken)> or accept an inbound stream using <xref:System.Net.Quic.QuicConnection.AcceptInboundStreamAsync(System.Threading.CancellationToken)>.
 
+See <see href="/dotnet/core/extensions/httpclient-http3#platform-dependencies">Platform dependencies</see> for QUIC prerequisities and supported operating systems.
```
Comment (gewarren) [inline xml/System.Net.Quic/QuicConnection.xml:31]:
```suggestion
For QUIC prerequisites and supported operating systems, see [Platform dependencies](/dotnet/core/extensions/httpclient-http3#platform-dependencies).
```

File: xml/System.Net.Quic/QuicConnection.xml [hunk 2]
```diff
@@ -161,12 +163,10 @@ If <xref:System.Net.Quic.QuicConnection.CloseAsync(System.Int64,System.Threading
...
       </ReturnValue>
       <Docs>
-        <summary>Gets a value that indicates whether QUIC is supported on the current machine.</summary>
+        <summary>Gets a value that indicates whether QUIC is supported for client scenarios on the current machine.</summary>
         <value>
           <see langword="true" /> if QUIC is supported on the current machine and can be used; otherwise, <see langword="false" />.</value>
-        <remarks>The current implementation depends on <see href="https://github.com/microsoft/msquic">MsQuic</see> native library, this property checks its presence (Linux machines).
-
-It also checks whether TLS 1.3, requirement for QUIC protocol, is available and enabled (Windows machines).</remarks>
+        <remarks>See <see href="/dotnet/core/extensions/httpclient-http3#platform-dependencies">Platform dependencies</see> for QUIC prerequisities and supported operating systems.</remarks>
```
Comment (gewarren) [inline xml/System.Net.Quic/QuicConnection.xml:169]:
```suggestion
        <remarks>For QUIC prerequisites and supported operating systems, see <see href="/dotnet/core/extensions/httpclient-http3#platform-dependencies">Platform dependencies</see>.</remarks>
```

File: xml/System.Net.Quic/QuicListener.xml [hunk 3]
```diff
@@ -106,12 +108,10 @@ This method propagates exceptions from <xref:System.Net.Quic.QuicListenerOptions
...
       </ReturnValue>
       <Docs>
-        <summary>Gets a value that indicates whether QUIC is supported on the current machine.</summary>
+        <summary>Gets a value that indicates whether QUIC is supported for server scenarios on the current machine.</summary>
         <value>
           <see langword="true" /> if QUIC is supported on the current machine and can be used; otherwise, <see langword="false" />.</value>
-        <remarks>The current implementation depends on <see href="https://github.com/microsoft/msquic">MsQuic</see> native library, this property checks its presence (Linux machines).
-
-It also checks whether TLS 1.3, requirement for QUIC protocol, is available and enabled (Windows machines).</remarks>
+        <remarks>See <see href="/dotnet/core/extensions/httpclient-http3#platform-dependencies">Platform dependencies</see> for QUIC prerequisities and supported operating systems.</remarks>
```
Comment (gewarren) [inline xml/System.Net.Quic/QuicListener.xml:114]:
```suggestion
        <remarks>For QUIC prerequisites and supported operating systems, see <see href="/dotnet/core/extensions/httpclient-http3#platform-dependencies">Platform dependencies</see>.</remarks>
```

File: xml/System.Net.Quic/QuicListener.xml [hunk 4]
```diff
@@ -26,6 +26,8 @@
 
 Unlike the connection and stream, the lifetime of a <xref:System.Net.Quic.QuicListener> isn't linked to any of the accepted connections. It can be safely disposed while keeping the accepted connection alive. Calling <xref:System.Net.Quic.QuicListener.DisposeAsync> will only cause the listener to stop listening for any other inbound connections.
 
+See <see href="/dotnet/core/extensions/httpclient-http3#platform-dependencies">Platform dependencies</see> for QUIC prerequisities and supported operating systems.
```
Comment (gewarren) [inline xml/System.Net.Quic/QuicListener.xml:29]:
```suggestion
For QUIC prerequisites and supported operating systems, see [Platform dependencies](/dotnet/core/extensions/httpclient-http3#platform-dependencies).
```

File: xml/System.Net.Quic/QuicConnection.xml (hunk 1 above)
Comment (gewarren) [reply xml/System.Net.Quic/QuicConnection.xml:31]: This is a Markdown section.

**Review by wfurt — APPROVED** [review]
LGTM. 
We should perhaps also improve the linked page but that can be separate change. 

Windows should mention Server 2022 IMHO.
Linux calls for OpenSSL 1.1 and that seems incomplete. There is 1.1.0 and 1.1.1. First one will not work (and is out of support) We should probably explicitly mention that OpenSSL 3 is NOT supported e.g. we will have problems on (.NET supported) Ubuntu 22  
Perhaps link to MsQuic support page. 

Also Pointing Quic to HTTP3 seems like reverse order since Http3 depends on Quic not vice versa.

**Comment by jkotas** [comment]
> Windows should mention Server 2022 IMHO.

I will update the other PR with this info.

> There is 1.1.0 and 1.1.1. First one will not work (and is out of support) We should probably explicitly mention that OpenSSL 3 is NOT supported
> Perhaps link to MsQuic support page.

Do you mean https://github.com/microsoft/msquic/blob/main/docs/Platforms.md ? It does not have any of these details about OpenSSL.

> Also Pointing Quic to HTTP3 seems like reverse order since Http3 depends on Quic not vice versa.

I agree. Unfortunately, we have linked to the HTTP3 doc for QUIC platform support from number of places already (blog posts, issues, etc.).

**Comment by ManickaP** [comment]
Eventually, there should be conceptual docs for System.Net.Quic. We should then shuffle it and put the prereqs, dependecies, msquic docs links etc. there instead.
I would keep it as it is now, and fix this all with S.N.Quic conceptual docs for which I created an issue: https://github.com/dotnet/docs/issues/33287

**Review by ManickaP — APPROVED** [review]
Thanks!

**Pull request merged**